src/overlays/actors/ovl_En_In/z_en_in.c

```diff
@@ -21,92 +44,1623 @@ const ActorInit En_In_InitVars = {
+                                            this->actor.world.pos.y, this->actor.world.pos.z, this->actor.shape.rot.x,
+                                            this->actor.shape.rot.y, this->actor.shape.rot.z, 0xF);
+                                Actor_MarkForDeath(&this->actor);
+                            } else {
+                                func_808F30B0(&this->skelAnime, 0);
+                                this->actionFunc = func_808F5A94;
+                            }
+                        }
+                    } else {
+                        if (gSaveContext.weekEventReg[52] & 1) {
+                            Actor_MarkForDeath(&this->actor);
+                        } else {
+                            func_808F30B0(&this->skelAnime, 7);
+                            this->actionFunc = func_808F5B58;
+                        }
+                    }
+                } else {
+                    Actor_MarkForDeath(&this->actor);
+                }
+            }
+        } else {
+            this->actionFunc = func_808F3618;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_In_0x808F30B0/func_808F4414.asm")
+void EnIn_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnIn* this = THIS;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_In_0x808F30B0/func_808F5674.asm")
+    Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_In_0x808F30B0/func_808F5728.asm")
+void EnIn_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnIn* this = THIS;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_In_0x808F30B0/func_808F5994.asm")
+    func_808F3310(this, globalCtx);
+    func_808F3334(this, globalCtx);
+    if (this->unk4AC & 2) {
+        this->unk4AC &= ~2;
+        func_808F38F8(this, globalCtx);
+    }
+    if (func_8012403C(globalCtx) == 8) {
+        this->unk4AC |= 0x40;
+    } else {
+        this->unk4AC &= ~0x40;
+    }
+    this->actionFunc(this, globalCtx);
+    func_800B78B8(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    func_808F3414(this, globalCtx);
+    func_808F32A0(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_In_0x808F30B0/func_808F5A34.asm")
+void func_808F6334(EnIn* this, GlobalContext* globalCtx) {
+    ActorPlayer* player = PLAYER;
+    s32 newUnk4C8;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_In_0x808F30B0/func_808F5A94.asm")
+    newUnk4C8 = func_80152498(&globalCtx->msgCtx);
+    this->unk4C4 += this->unk4C0 != 0.0f ? 40.0f : -40.0f;
+    this->unk4C4 = CLAMP(this->unk4C4, 0.0f, 80.0f);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_In_0x808F30B0/func_808F5B58.asm")
+    SysMatrix_InsertTranslation(this->unk4C4, 0.0f, 0.0f, MTXMODE_APPLY);
+    if (this == (EnIn*)player->unkA88 && !(globalCtx->msgCtx.unk11F04 >= 0xFF && globalCtx->msgCtx.unk11F04 < 0x201) &&
```

Comment:
Are you sure this player ptr which uses the PLAYER macro is then converted to En_In? wouldn't this be a player+0xA88 lookup? 0xA88 is an actor ptr in ActorPlayer struct:
https://github.com/zeldaret/mm/blob/master/include/z64actor.h#L251